features/steps/data.py

```diff
@@ -115,9 +115,10 @@ def step_impl(context, correct_metadata):
     correct_metadata_file = open(correct_metadata_path)
     correct_metadata_json = json.load(correct_metadata_file)
 
+    result = diff(context.json_output,correct_metadata_json)
     assert (
         context.json_output == correct_metadata_json
-    ), f"Metadata does not match expected metadata from {correct_metadata_json}."
+    ), f"Metadata does not match expected metadata, the following values are missing from the metadat output :\n {list(result)}."
```

Comment:
Add something to the error message to print out the full metadata output alongside the `diff` result, with info on interpreting the error (e.g. `add` means something is missing. `remove` means something needs deleting etc)

Reply:
<img width="1458" alt="Screenshot 2024-04-24 at 08 35 17" src="https://github.com/ONSdigital/dp-data-pipelines/assets/43893201/7a233796-0c49-4231-bb26-84c841139e5f">
Changed the message and now with proper data it shows the path to the data that needs changing(or missing).